Make toEqual matcher report the difference between arrays with different length

kiewic opened the pull request:
The purpose of this change, is to improve the readability of specs that fail because two arrays have different length. Instead of printing the regular error message:

```bash
Expected $.values = [ 'foo', 'bar', 'baz', 'fizz', 'buzz', 'bazz', 'fur', 'bliz', 'pugg', 'bizz', 'fixx' ] to equal [ 'foo', 'bar', 'baz', 'fizz', 'buzz', 'bazz', 'fur', 'pugg', 'bizz' ].
```

The error message will include only the differences between the two arrays:

```bash
Expected $.values[7] = 'bliz' to equal 'pugg'.
Expected $.values[8] = 'pugg' to equal 'bizz'.
Expected $.values[9] = 'bizz' to equal undefined.
Expected $.values[10] = 'fixx' to equal undefined.
```

Please take a look.

sgravrock commented:
This looks like a potentially useful change, but I'm seeing a couple of issues with the PR as it stands now:

* It breaks some of the existing tests.
* It treats different-length arrays as equal if the extra elements are all `undefined`. For instance, `expect([1, 2, undefined]).toEqual([1, 2])` should fail but this PR makes it pass.

I'd be happy to review a version of this that passes all tests and changes only the message produced by the matcher, not its pass/fail behavior. Thanks for using Jasmine!

kiewic commented:
Hi @sgravrock, thank you for looking at my PR. I put back the array length comparison, and added a test to cover the case you described. The errors messages now include a line like `Expected $.length = 1 to equal 2.` to clarify the error, although this may be redundant. I'm happy to hear your suggestions.

Also reverted changes to *jasmine.js* as described in the instructions.

Please take a look again.
